src/components/FeaturePanel/Climbing/ClimbingStructuredData.tsx

```diff
@@ -13,105 +13,85 @@ import {
 } from './utils/grades/routeGrade';
 import { useUserSettingsContext } from '../../utils/UserSettingsContext';
 
-export const ClimbingStructuredData = () => {
-  const { feature } = useFeatureContext();
+const generateScriptContent = (feature, userSettings) => {
   const isClimbingArea = feature.tags.climbing === 'area';
   const isClimbingCrag = feature.tags.climbing === 'crag';
   const isClimbingRoute = feature.tags.climbing === 'route_bottom';
   const poiType = getHumanPoiType(feature);
-  const { userSettings } = useUserSettingsContext();
 
   if (isClimbingArea) {
-    return (
-      <Script
-        id="structured-data-climbing-area"
-        type="application/ld+json"
-        dangerouslySetInnerHTML={{
-          __html: `{
+    return `{
       "@context": "https://schema.org",
       "@type": "Place",
       "name": "${feature.tags.name} - ${poiType}",
-      ${feature.tags.description ? `"description": "${feature.tags.description}"` : ''},
+      ${feature.tags.description ? `"description": "${feature.tags.description}",` : ''}
       "geo": {
         "@type": "GeoCoordinates",
         "latitude": "${feature.center[1]}",
         "longitude": "${feature.center[0]}"
       }
-    }`,
-        }}
-      />
-    );
+    }`;
   }
 
   if (isClimbingCrag) {
     const photoPaths = getWikimediaCommonsPhotoKeys(feature.tags);
-
-    return (
-      <Script
-        id="structured-data-climbing-crag"
-        type="application/ld+json"
-        dangerouslySetInnerHTML={{
-          __html: `{
+    return `{
       "@context": "https://schema.org",
       "@type": "Place",
       "name": "${feature.tags.name} - ${poiType}",
-      ${
-        photoPaths.length > 0
-          ? `"image": [
-        ${photoPaths.map((photoPath) => `"${getCommonsImageUrl(feature.tags[photoPath], 800)}"`).join(', ')}
-      ],`
-          : ''
-      }
-      ${feature.tags.description ? `"description": "${feature.tags.description}"` : ''},
+      ${photoPaths.length > 0 ? `"image": [${photoPaths.map((photoPath) => `"${getCommonsImageUrl(feature.tags[photoPath], 800)}"`).join(', ')}],` : ''}
+      ${feature.tags.description ? `"description": "${feature.tags.description}",` : ''}
       "geo": {
         "@type": "GeoCoordinates",
         "latitude": "${feature.center[1]}",
         "longitude": "${feature.center[0]}"
       }
-    }`,
-        }}
-      />
-    );
+    }`;
   }
 
   if (isClimbingRoute) {
     const photoPaths = getWikimediaCommonsPhotoKeys(feature.tags);
     const selectedRouteSystem = userSettings['climbing.gradeSystem'];
     const routeDifficulties = getDifficulties(feature.tags);
-
     const { routeDifficulty } = findOrConvertRouteGrade(
       routeDifficulties,
       selectedRouteSystem,
     );
 
-    return (
-      <Script
-        id="structured-data-climbing-route"
-        type="application/ld+json"
-        dangerouslySetInnerHTML={{
-          __html: `{
+    return `{
```

Comment:
What about using a real JSON and then paste it inside the `<Script>` with `JSON.stringify`? This string kung-fu looks prone to error. 😅